oap-server/server-query-plugin/query-graphql-plugin/src/main/java/org/apache/skywalking/oap/query/graphql/resolver/OndemandLogQuery.java

```diff
@@ -34,64 +34,87 @@
-            return nsList
-                .getItems()
-                .stream()
-                .map(V1Namespace::getMetadata)
-                .filter(Objects::nonNull)
-                .map(V1ObjectMeta::getName)
-                .collect(Collectors.toList());
-        } catch (ApiException e) {
-            log.error("Failed to list namespaces from Kubernetes, {}", e.getResponseBody(), e);
+    private final MetadataQueryV2 metadataQuery;
 
-            Map<String, Object> responseBody = gson.fromJson(e.getResponseBody(), responseType);
-            String message = responseBody.getOrDefault("message", e.getCode()).toString();
-            throw new RuntimeException(message);
-        }
+    public List<String> listContainers(final OndemandContainergQueryCondition condition)
+        throws IOException {
+        final ServiceInstance instance =
+            metadataQuery.getInstance(condition.getServiceInstanceId());
+        final Map<String, String> attributesMap = convertInstancePropertiesToMap(instance);
+        final String ns = attributesMap.get(PropertyUtil.NAMESPACE);
+        final String pod = attributesMap.get(PropertyUtil.POD);
+        return listContainers(ns, pod);
     }
 
-    public List<String> listContainers(final OndemandLogQueryCondition condition)
+    public Logs ondemandPodLogs(final OndemandLogQueryCondition condition)
         throws IOException {
-        final String ns = condition.getNamespace();
-        final IDManager.ServiceInstanceID.InstanceIDDefinition instanceIDDefinition =
-            IDManager.ServiceInstanceID.analysisId(condition.getServiceInstanceId());
-        final String instanceName = instanceIDDefinition.getName();
+        final ServiceInstance instance =
+            metadataQuery.getInstance(condition.getServiceInstanceId());
+        final Map<String, String> attributesMap = convertInstancePropertiesToMap(instance);
+        final String ns = attributesMap.get(PropertyUtil.NAMESPACE);
+        final String pod = attributesMap.get(PropertyUtil.POD);
+        return ondemandPodLogs(ns, pod, condition);
+    }
 
+    protected Map<String, String> convertInstancePropertiesToMap(final ServiceInstance instance) {
+        if (instance == null) {
+            return Collections.emptyMap();
+        }
+        final List<Attribute> attributes = instance.getAttributes();
+        if (attributes == null || attributes.isEmpty()) {
+            return Collections.emptyMap();
+        }
+        final Map<String, String> attributesMap =
+            attributes
+                .stream()
+                .collect(Collectors.toMap(Attribute::getName, Attribute::getValue));
+        if (!attributesMap.containsKey(PropertyUtil.NAMESPACE)
+            || !attributesMap.containsKey(PropertyUtil.POD)) {
+            return Collections.emptyMap();
+        }
+        return attributesMap;
+    }
+
+    public List<String> listContainers(
```

Comment:
@Fine0830 Once UI faces `listContainers` returning an empty list, please show a tip on the logs page, **The on-demand logs are not available.**